flink-connector-tidb-cdc/src/main/java/com/ververica/cdc/connectors/tidb/table/TiDBTableSourceFactory.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.tidb.table;
+
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.connector.source.DynamicTableSource;
+import org.apache.flink.table.factories.DynamicTableSourceFactory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.utils.TableSchemaUtils;
+
+import com.ververica.cdc.connectors.tidb.TDBSourceOptions;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.ververica.cdc.connectors.tidb.TDBSourceOptions.DATABASE_NAME;
+import static com.ververica.cdc.connectors.tidb.TDBSourceOptions.HOSTNAME;
+import static com.ververica.cdc.connectors.tidb.TDBSourceOptions.PASSWORD;
+import static com.ververica.cdc.connectors.tidb.TDBSourceOptions.SCAN_STARTUP_MODE;
+import static com.ververica.cdc.connectors.tidb.TDBSourceOptions.TABLE_NAME;
+import static com.ververica.cdc.connectors.tidb.TDBSourceOptions.USERNAME;
+
+/** Factory for creating configured instance of {@link TiDBTableSource}. */
+public class TiDBTableSourceFactory implements DynamicTableSourceFactory {
+
+    private static final String IDENTIFIER = "tidb-cdc";
+
+    @Override
+    public DynamicTableSource createDynamicTableSource(Context context) {
+        final FactoryUtil.TableFactoryHelper helper =
+                FactoryUtil.createTableFactoryHelper(this, context);
+
+        final ReadableConfig config = helper.getOptions();
+        String hostname = config.get(HOSTNAME);
```

Comment:
Option `TDBSourceOptions.TIKV_PD_ADDRESSES` is required,but you do not add to `TiDBTableSource`‘s constructor parameter.